overlay-debs/mesa-snapshot/mesa-snapshot.sh

```diff
@@ -0,0 +1,52 @@
+#! /bin/sh
+#
+# Copyright (c) Qualcomm Technologies, Inc. and/or its subsidiaries.
+# SPDX-License-Identifier: BSD-3-Clause
+
+COMMIT=${COMMIT:-origin/main}
+
+set -e
+
+DEB_FILE=${DSC_FILE%%.dsc}.debian.tar.xz
+ORIG_FILE=$(echo ${DSC_FILE} | sed -e 's/-.*/.orig.tar.xz/')
+
+rm -rf mesa
+git clone --depth 1 https://gitlab.freedesktop.org/mesa/mesa
+
+cd mesa
+git fetch --depth 1 origin ${COMMIT##origin/}
+
+date=$(git log -1 --format=%cd --date=format:%Y%m%d ${COMMIT})
+subject=$(git log -1 --format="%h (\"%s\")" ${COMMIT})
+version=25.2.0~git${date}
+
+rm -rf ../mesa-${version}
+mkdir ../mesa-${version}
+git archive --format=tar HEAD | tar x -C ../mesa-${version}
+
+cd ../mesa-${version}
+
+rm -rf debian
+tar xJf ${DEB_FILE}
+
+cat >> debian/changelog.tmp << EOF
+mesa (${version}-0) experimental; urgency=medium
```

Comment:
would you mind changing the version to 0qcom1 so that all packages with qcom changes have a qcom version?